### PR TITLE
Optimise `cached_head_ptr` calls within process block

### LINKED ISSUE #4972: Break down `process_block` section in `deployment_sync_secs` metrics to smaller sections
### Description

_No response_

### Are you aware of any blockers that must be resolved before implementing this feature? If so, which? Link to any relevant GitHub issues.

_No response_

### Some information to help us out

- [ ] Tick this box if you plan on implementing this feature yourself.
- [X] I have searched the issue tracker to make sure this issue is not a duplicate.

### PR DESCRIPTION
**Pull request by incrypto32**
Closes #4972

### CODE REVIEW

**Comment by incrypto32** [comment]
Let me know if the sections for process_block could be better, i was unclear how to divide it into more sections

File: core/src/subgraph/runner.rs [hunk 1]
```diff
@@ -899,12 +900,14 @@ where
                 .observe(block.trigger_count() as f64);
         }
 
+        let ptr_updates_section = self.metrics.stream.stopwatch.start_section("ptr_updates");
+
         if block.trigger_count() == 0
```
Comment (leoyvens) [inline core/src/subgraph/runner.rs:905]: An idea, lets wrap this whole if block in an outer if with the condition:
`if self.state.skip_ptr_updates_timer.elapsed() && !self.state.synced`
This way, if the timer hasn't elapsed or the subgraph is synced, we won't even bother with resetting the timer and updating the cached head.

Comment (incrypto32) [reply core/src/subgraph/runner.rs]: Just pushed a commit, hope the change made is what you suggested

Comment (incrypto32) [reply core/src/subgraph/runner.rs]: hmm, intereseting i thought when the variable that is assigned to the section goes out of scope the timer is reset.

Comment (zorancv) [reply core/src/subgraph/runner.rs]: I think that the check for `!self.state.synced` should be separated and outer one, and then the else branch of the check for `self.state.skip_ptr_updates_timer.elapsed()` should reset the timer too.

**Review by zorancv — APPROVED** [review]
Other than the timer handling I've commented it looks good to me.

**Pull request merged**